tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/Unroller.scala

```diff
@@ -65,20 +64,22 @@ class Unroller( tracker : TransformationTracker ) extends TlaModuleTransformatio
     bodyMap.get( unrollLimitOperName ) match {
       case Some( unrollLimitDecl ) =>
         // The unrollLimit operator must not be recursive ...
+        val msg = s"Expected an integer bound in $unrollLimitOperName, found a recursive operator. See: " + MANUAL_LINK
         if ( unrollLimitDecl.isRecursive )
-          FailWith( new Exception( s"Unroll-limit operator $unrollLimitOperName is recursive." ) )
+          FailWith( new TlaInputError( msg ) )
         else
```

Comment:
Since `msg` is only used in one branch below, maybe best to just put it directly above the use site (as you do on line 81)?

Reply:
This is interesting. Idea did not want to introduce curly braces when one writes: `if (...) a else b`. That is another reason for always using curly braces.

Reply:
lol! 

I can't wait for the day we're all doing structured editing :D

Reply:
I have a childhood trauma with structured editing. I think it was QBasic that enforced structured editing instead of having a normal parser :D